src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs

```diff
@@ -102,21 +104,16 @@ internal static partial class SyntaxGeneratorExtensions
                 // Special case.  If we're generating a struct constructor, then we'll need
                 // to initialize all fields in the struct, not just the ones we're creating.  To
                 // do that, we call the default constructor.
-                var realFields = containingTypeOpt.GetMembers()
-                                     .OfType<IFieldSymbol>()
-                                     .Where(f => !f.IsStatic);
-                var initializedFields = parameterToExistingFieldMap.Values
-                                            .OfType<IFieldSymbol>()
-                                            .Where(f => !f.IsImplicitlyDeclared && !f.IsStatic);
-                if (initializedFields.Count() < realFields.Count())
-                {
-                    // We have less field assignments than actual fields.  Generate a call to the
-                    // default constructor as well.
-                    return ImmutableArray<SyntaxNode>.Empty;
-                }
+
+                return containingTypeOpt.GetMembers()
```

Comment:
probably want to remove the above comment adn replace witha  suitable new one explaining what's going on.  i.e. "if this is a struct constructor, we'll normally need to call the implicit `this()` constructor to initialize all fields.  However, we only need to do this if there's at least one instance field/property that isn't being initialized by some parameter to htis constructor."

Something like that.